### ISSUE TITLE
[@loopback/sequelize] Attempting to use `options` in a `where` filter leads to "There is no equivalent operator" exception

### ISSUE THREAD

**Issue by KalleV** [loopbackio/loopback-next#9690]
### Describe the bug

This section of the code throws an exception if the "where" filter includes an "options" key: https://github.com/loopbackio/loopback-next/blob/96b9a56788937f081e61f1d48dc080c7233d394c/extensions/sequelize/src/sequelize/sequelize.repository.base.ts#L604

Example filter that will throw an exception:
```
where: {
   name: {
     like: '%abcdefg%',
     options: 'i'
   }
}
```

References:
 - https://loopback.io/doc/en/lb4/Where-filter.html#operators

### Reproduction

N/A

**Comment by shubhamp-sf** [comment]
`@loopback/sequelize` is meant to be used with SQL dialects, this option being only supported by LoopBack in its memory and MongoDB connectors is not expected to work correctly. 
<img width="870" alt="Screenshot 2023-06-25 at 12 26 08 PM" src="https://github.com/loopbackio/loopback-next/assets/110156023/c990ecc5-883b-45d9-8c6e-166a9cbc0858">

Please use `ilike` or `regexp` operator instead depending on your needs.

**Comment by KalleV** [comment]
@shubhamp-sf That’s true but the existing connector doesn’t throw an exception even if it is unused by the underlying MySQL connector. I was thinking for backwards compatibility / migration purposes it could be a no-op.
